src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,566 @@
+        throw(ArgumentError("Constructing InterfaceValues with a single FaceQuadratureRule isn't valid for mixed grids, please consider passing quad_rule_there"))
+    here = FaceValues(quad_rule_here, ip_here, geo_ip_here)
+    there = FaceValues(quad_rule_there, ip_there, geo_ip_there)
+    return InterfaceValues{typeof(here), typeof(there)}(here, there)
+end
+
+InterfaceValues(facevalues_here::FVA, facevalues_there::FVB = deepcopy(facevalues_here)) where {FVA <: FaceValues, FVB <: FaceValues} =
+    InterfaceValues{FVA,FVB}(facevalues_here, facevalues_there)
+
+function getnbasefunctions(iv::InterfaceValues)
+    return getnbasefunctions(iv.here) + getnbasefunctions(iv.there)
+end
+# function getngeobasefunctions(iv::InterfaceValues)
+#     return getngeobasefunctions(iv.here) + getngeobasefunctions(iv.there)
+# end
+
+"""
+    getnquadpoints(iv::InterfaceValues)
+
+Return the number of quadrature points on the interface for the current (most recently
+[`reinit!`](@ref)ed) interface.
+"""
+getnquadpoints(iv::InterfaceValues) = getnquadpoints(iv.here)
+
+@propagate_inbounds function getdetJdV(iv::InterfaceValues, q_point::Int)
+    return getdetJdV(iv.here, q_point)
+end
+
+"""
+    reinit!(
+        iv::InterfaceValues,
+        cell_here::AbstractCell, coords_here::AbstractVector{Vec{dim, T}}, face_here::Int,
+        cell_there::AbstractCell, coords_there::AbstractVector{Vec{dim, T}}, face_there::Int
+    )
+
+Update the [`InterfaceValues`](@ref) for the interface between `cell_here` (with cell
+coordinates `coords_here`) and `cell_there` (with cell coordinates `coords_there`).
+`face_here` and `face_there` are the (local) face numbers for the respective cell.
+"""
+function reinit!(
+        iv::InterfaceValues,
+        cell_here::AbstractCell, coords_here::AbstractVector{Vec{dim, T}}, face_here::Int,
+        cell_there::AbstractCell, coords_there::AbstractVector{Vec{dim, T}}, face_there::Int
+    ) where {dim, T}
+
+    # reinit! the here side as normal
+    reinit!(iv.here, coords_here, face_here)
+    dim == 1 && return reinit!(iv.there, coords_there, face_there)
+    # Transform the quadrature points from the here side to the there side
+    iv.there.current_face[] = face_there
+    interface_transformation = InterfaceTransformation(cell_here, cell_there, face_here, face_there)
+    quad_points_a = getpoints(iv.here.qr, face_here)
+    quad_points_b = getpoints(iv.there.qr, face_there)
+    transform_interface_points!(quad_points_b, quad_points_a, interface_transformation)
+    @boundscheck checkface(iv.there, face_there)
+    # TODO: This is the bottleneck, cache it?
+    @assert length(quad_points_a) <= length(quad_points_b)
+    # Re-evalutate shape functions in the transformed quadrature points
+    for qp in 1:length(quad_points_a), i in 1:getnbasefunctions(iv.there)
+        iv.there.dNdξ[i, qp, face_there], iv.there.N[i, qp, face_there] = shape_gradient_and_value(iv.there.func_interp, quad_points_b[qp], i)
```

Comment:
If we are shuffling around the shape values on the "there"-side, does this not mean that we have to shuffle the "interfacedof" vector on the there side aswell? (i.e. `interfacedofs = [cellhere_dofs, cellthere_dofs]` would need to be reordered aswell for the assembly in the the siffness matrix).
Would it not be better to reorder the quad_points_b such that they match quad_points_a instead, and keep the order of the shape-functions fixed?

I might be wrong though...

Reply:
There is no reordering of the shape functions, just re-evaluation on new quadrature points.

Reply:
Yeah, it's just transforming the quadrature points to match coordinates, shape functions order should remain the same (1:nbasefunc in the `there` interpolation)

Reply:
I propose we can later in separate PRs do such optimizations via specialized dispatches of special cases if we want, but let us for now take a solution which works in more general scenarios.